core/src/main/java/com/rudderstack/android/sdk/core/RudderDeviceModeManager.java

```diff
@@ -277,11 +298,39 @@ void dumpTransformedEvents(TransformationResponse transformationResponse) {
                 return;
             List<TransformedEvent> transformedEvents = transformedDestination.payload;
             sortTransformedEventBasedOnOrderNo(transformedEvents);
-            for (TransformedEvent transformedEvent : transformedEvents) {
-                if (transformedEvent.status.equals("200")) {
-                    dumpEventToDestinations(transformedEvent.event, Collections.singletonList(destinationName), "dumpTransformedEvents");
+            sendEventsToTransformedDestinations(transformedDestination, destinationName);
+        }
+    }
+
+    private void sendEventsToTransformedDestinations(TransformedDestination transformedDestination, String destinationName) {
+        for (TransformedEvent transformedEvent : transformedDestination.payload) {
+            RudderMessage message = transformedEvent.event;
+            boolean onTransformationError = !transformedEvent.status.equals("200");
+            if (onTransformationError) {
+                StringBuilder errorMsg = new StringBuilder();
+                errorMsg.append("RudderDeviceModeManager: dumpTransformedEvents: ");
+                if (transformedEvent.status.equals("410")) {
+                    errorMsg.append("The requested transformation is not available on the destination or there is a configuration issue. ");
+                } else {
+                    // For all other status codes
+                    errorMsg.append("There is a transformation error. ");
+                }
+                // If their is a transformation error then response payload will not contain the original event. So we need to get the original event based on messageId/orderNo.
+                message = this.deviceModeTransformationManager.getEventFromMessageId(transformedEvent.orderNo);
+                if (destinationsExcludedOnTransformationError.contains(destinationName)) {
+                    errorMsg.append(destinationName).append(" is excluded from accepting event ").append(message.getEventName()).append(" on transformation error. Hence dropping this event.");
+                    RudderLogger.logWarn(errorMsg.toString());
+                    continue;
+                } else {
+                    errorMsg.append("Sending the untransformed event ").append(message.getEventName());
+                    RudderLogger.logWarn(errorMsg.toString());
```

Comment:
are we doing this for all error status codes?

Reply:
Yeah if there is any kind of event-level transformation issues/error, we are doing this (i.e., logging this error). 
Let me know if you need to know something else.